server.go

```diff
@@ -266,6 +266,29 @@ func CustomCodec(codec Codec) ServerOption {
 	})
 }
 
+// ForcedCodec returns a ServerOption that sets a codec for message marshaling and unmarshaling.
+//
+// This will override any lookups by content-subtype for Codecs registered with RegisterCodec.
+//
+// See Content-Type on
+// https://github.com/grpc/grpc/blob/master/doc/PROTOCOL-HTTP2.md#requests for
+// more details. Also see the documentation on RegisterCodec and
+// CallContentSubtype for more details on the interaction between encoding.Codec and
+// content-subtype.
+//
+// This function is provided for advanced users; prefer to use only
+// CallContentSubtype to select a registered codec on the client instead.
+//
+// Experimental
+//
+// Notice: This API is EXPERIMENTAL and may be changed or removed in a
+// later release.
+func ForcedCodec(codec encoding.Codec) ServerOption {
+	return newFuncServerOption(func(o *serverOptions) {
+		o.codec = codec
```

Comment:
Well this was surprisingly simple.  :)

Could you please add tests in `test/end2end_test.go` for this?  See https://github.com/grpc/grpc-go/blob/4f3aa7cfa157c38bd5c2da7f4568614f815ab4ad/test/end2end_test.go#L5260 for an example of the style of test we use now.  (`Start` accepts `ServerOption`s already, so it should be pretty simple to get working.)